### PR TITLE
Fix ShadowMap NaNs

### PR DESCRIPTION
**Pull request by lilleyse**
For https://github.com/AnalyticalGraphicsInc/cesium/pull/3605

@hpinkos 

This fixes the NaNs.

Terrain shadows are still broken after https://github.com/AnalyticalGraphicsInc/cesium/pull/4339 was merged. I'm still investigating this but wanted to get the NaN fixes in the meantime.

### CODE REVIEW

File: Source/Scene/ShadowMap.js [hunk 1]
```diff
@@ -1317,7 +1317,7 @@ define([
         if (shadowMap._fitNearFar) {
             // shadowFar can be very large, so limit to shadowMap.maximumDistance
             near = Math.min(frameState.shadowHints.nearPlane, shadowMap.maximumDistance);
-            far = Math.min(frameState.shadowHints.farPlane, shadowMap.maximumDistance);
+            far = Math.min(frameState.shadowHints.farPlane, shadowMap.maximumDistance + 1.0);
```
Comment (pjcozzi) [inline Source/Scene/ShadowMap.js]: Might need a comment why here.

**Comment by pjcozzi** [comment]
@hpinkos please merge when ready.

**Comment by lilleyse** [comment]
Updated with comment.

**Comment by hpinkos** [comment]
Great, thanks @lilleyse!

Did you open an issue for the terrain shadow problem?  If not, please do so.

**Pull request merged**